haddock-api/src/Haddock/Interface/Create.hs

```diff
@@ -343,7 +344,7 @@ parseOption "prune"           = return (Just OptPrune)
 parseOption "ignore-exports"  = return (Just OptIgnoreExports)
 parseOption "not-home"        = return (Just OptNotHome)
 parseOption "show-extensions" = return (Just OptShowExtensions)
-parseOption other = tell ["Unrecognised option: " ++ other] >> return Nothing
+parseOption other = tell [dislocatedErrMsg $ "Unrecognised option: " ++ other] >> return Nothing
```

Comment:
This error message should look almost exactly like the `mkDocOpts` one since they are referring to the same options. That includes passing in the same module location (instead of `dislocatedErrMsg`).